spritecontrol.cpp

```diff
@@ -101,7 +105,8 @@ bool SpriteChoreographer::should_change_pattern() {
 }
 
 void SpriteChoreographer::change_pattern() {
-	m_pattern = (PatternName) (Noise::random() * cast<double>(_PATTERN_COUNT));
+	size_t rand = (size_t) (Noise::random() * m_enabled_patterns.size());
+	m_pattern = m_enabled_patterns.at(rand);
```

Comment:
I think we should make it so that the choreographer can never change into the same pattern it's already playing, since that behavior feels pretty unintuitive. If I ask for two patterns I would expect it to just swap between those patterns, for example.

Reply:
Yeah, probably. As far as I'm aware changing into the same pattern was always possible, but it's definitely a lot more noticeable with 2 patterns than with 8.